modules/ocl/cl_utils.cpp

```diff
@@ -86,7 +86,19 @@ convert_to_climage (
     SmartPtr<CLVideoBuffer> cl_video_buf = buf.dynamic_cast_ptr<CLVideoBuffer> ();
     if (cl_video_buf.ptr ()) {
         SmartPtr<CLBuffer> cl_buf = cl_video_buf->get_cl_buffer ();
-        cl_image = new CLImage2D (context, desc, flags, cl_buf);
+        SmartPtr<CLBuffer> main_buf;
+
+        if (offset == 0) {
+            main_buf = cl_buf;
+        } else {
+            uint32_t row_pitch = CLImage::calculate_pixel_bytes (desc.format) *
+                                     XCAM_ALIGN_UP (desc.width, XCAM_CL_IMAGE_ALIGNMENT_X);
+            uint32_t size = row_pitch * desc.height;
+
+            main_buf = new CLSubBuffer (context, cl_buf, flags, offset, size);
+        }
+
```

Comment:
get_cl_buffer returned a cl_buffer(main_buf) and lost the buffer proxy(cl_video_buf) which may release the buffer to pool. but the cl_buf still in use in Image2D.